src/sentry/snuba/metrics/datasource.py

```diff
@@ -368,6 +373,9 @@ def get_tag_values(
     """Get all known values for a specific tag"""
     assert projects
 
+    if tag_name in UNALLOWED_TAGS:
```

Comment:
I realize this may seem like a fickle solution but the alternative would be to traverse down the expression tree of each and every condition, and quite honestly it seems like an over kill. 

The event parsing grammar that matches with this `text_filter` expects a string directly followed by a separator, and so unless this grammar changes, we should be good on this check, and frankly I don't think it will because then it would affect many other places